Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py

```diff
@@ -551,6 +553,618 @@ def decode_url(headers: str) -> dict:
+            "requestHeaders": "User-Agent%3a%20BOT%2f0.1%20(BOT%20for%20JCE)%0d%0aAccept%3a%20text%2fhtml,application%2fxhtml+xml,application%2fxml%3bq%3d0.9,*%2f*%3bq%3d0.8%0d%0auniqueID%3a%20CR_H8%0d%0aAccept-Language%3a%20en-US,en%3bq%3d0.5%0d%0aAccept-Encoding%3a%20gzip,%20deflate%0d%0aConnection%3a%20keep-alive%0d%0aHost%3a%20www.hmapi.com%0d%0aContent-Length%3a%200%0d%0a",
+            "requestId": "1158db1758e37bfe67b7c09",
+            "responseHeaders": "Server%3a%20AkamaiGHost%0d%0aMime-Version%3a%201.0%0d%0aContent-Type%3a%20text%2fhtml%0d%0aContent-Length%3a%20266%0d%0aExpires%3a%20Tue,%2004%20Apr%202017%2010%3a57%3a02%20GMT%0d%0aDate%3a%20Tue,%2004%20Apr%202017%2010%3a57%3a02%20GMT%0d%0aConnection%3a%20close%0d%0aSet-Cookie%3a%20ak_bmsc%3dAFE4B6D8CEEDBD286FB10F37AC7B256617DB580D417F0000FE7BE3580429E23D%7epluPrgNmaBdJqOLZFwxqQLSkGGMy4zGMNXrpRIc1Md4qtsDfgjLCojg1hs2HC8JqaaB97QwQRR3YS1ulk+6e9Dbto0YASJAM909Ujbo6Qfyh1XpG0MniBzVbPMUV8oKhBLLPVSNCp0xXMnH8iXGZUHlUsHqWONt3+EGSbWUU320h4GKiGCJkig5r+hc6V1pi3tt7u3LglG3DloEilchdo8D7iu4lrvvAEzyYQI8Hao8M0%3d%3b%20expires%3dTue,%2004%20Apr%202017%2012%3a57%3a02%20GMT%3b%20max-age%3d7200%3b%20path%3d%2f%3b%20domain%3d.hmapi.com%3b%20HttpOnly%0d%0a",
+            "start": f"{int(time.time())}",
+            "status": "200"
+        },
+        "type": "akamai_siem",
+        "userRiskData": {
+            "allow": "0",
+            "general": "duc_1h:10|duc_1d:30",
+            "originUserId": "jsmith007",
+            "risk": "udfp:1325gdg4g4343g/M|unp:74256/H",
+            "score": "75",
+            "status": "0",
+            "trust": "ugp:US",
+            "username": "jsmith@example.com",
+            "uuid": "964d54b7-0821-413a-a4d6-8131770ec8d5"
+        },
+        "version": "1.0"
+        }
+        import copy
+        for i in range(300000):
+            duplicated_dict = copy.deepcopy(original_dict)
+            duplicated_dict["unique_id"] = i
+            events.append(json.dumps(duplicated_dict))
+        events.append(json.dumps({
+            "total": 300000,
+            "offset": "Hayun offset",
+            "limit": 300000
+        }))
+        EVENTS_LS = "\n".join(events)
+    else:
+        demisto.info("Skipping creating new events, going to sleep for 5 seconds.")
+        await asyncio.sleep(5)
+        demisto.info("Woke up after 5 seconds, continue running.")
+    return EVENTS_LS
+
+
+''' COMMANDS '''
+
+
+async def wait_until_tasks_load_decrees():
+    while (current_tasks_total_size := get_tasks_total_size()) > MAX_ALLOWED_TASKS_SIZE:
+        demisto.info(f"current tasks total size = {current_tasks_total_size} is larger than the max allowed tasks size"
+                     f"{MAX_ALLOWED_TASKS_SIZE}. sleeping for 60 seconds to let other tasks finish.")
+        await asyncio.sleep(60)
+
+
+def get_tasks_total_size() -> int:
+    tasks = asyncio.all_tasks()
+    total_size = 0
+    for task in tasks:
+        total_size += sys.getsizeof(task)
```

Comment:
check this doesnt add more than a quarter second on each run
double check the mechanism goes up in memory the more we have uncompleted tasks
(add asyncio.sleep(1 hour) to check)

Reply:
After doing some research, since getsizeof retrieving the size of the task object regardless of the data it contains - we've decided to count the number of active tasks rather than sum up the total size.